service/controller/azurecluster/handler/subnet/resource.go

```diff
@@ -406,7 +406,7 @@ func (r *Resource) isDeploymentOutOfDate(ctx context.Context, allocatedSubnet *c
 	return allocatedSubnet.Name != nodepoolName || allocatedSubnet.CIDRBlocks[0] != subnetCidr, nil
 }
 
-func (r *Resource) getDeploymentParameters(azureCluster *capzv1alpha3.AzureCluster, natGatewayId string, allocatedSubnet *capzv1alpha3.SubnetSpec) (map[string]interface{}, error) {
+func (r *Resource) getDeploymentParameters(azureCluster *capz.AzureCluster, natGatewayId string, allocatedSubnet capz.SubnetSpec) (map[string]interface{}, error) {
```

Comment:
I don't think we change the subnetspec here so it should be fine not using a pointer